### DISCUSSION TITLE
Probably unnecessary Array copy

### DISCUSSION THREAD

**Discussion by samuelorji** [playframework/playframework#12856] [General]
Hi, 

Is there any reason why we can't call `ByteString.fromArrayUnsafe` [here](https://github.com/playframework/playframework/blob/main/core/play/src/main/scala/play/api/http/Writeable.scala#L206) instead of calling `apply` on ByteString which should save us an [array copy ](https://github.com/apache/pekko/blob/main/actor/src/main/scala-2.13/org/apache/pekko/util/ByteString.scala#L31). 

We already do `ByteString.fromArrayUnsafe` for the `JsValue` writeable [here](https://github.com/playframework/playframework/blob/main/core/play/src/main/scala/play/api/http/Writeable.scala#L106).

**Comment by PromanSEW** [comment]
Please provide PR to fix this behaviour so :)

**Comment by maxcom** [comment]
Array is a mutable structure and it is not safe to assume that it won't change. `JsValue` case is different, because we know that `Json.toBytes` allocates new array on each call.
